
RCS file: /cvs/xc/programs/Xserver/hw/xfree86/common/xf86Config.c,v
retrieving revision 3.272
retrieving revision 3.273
diff -u -r3.272 -r3.273
--- xc/programs/Xserver/hw/xfree86/common/xf86Config.c	24 Aug 2003 20:52:30 -0000	3.272
+++ xc/programs/Xserver/hw/xfree86/common/xf86Config.c	27 Aug 2003 02:50:25 -0000	3.273
@@ -1,4 +1,4 @@
-/* $XFree86: xc/programs/Xserver/hw/xfree86/common/xf86Config.c,v 3.272 2003/08/24 20:52:30 dawes Exp $ */
+/* $XFree86: xc/programs/Xserver/hw/xfree86/common/xf86Config.c,v 3.273 2003/08/27 02:50:25 dawes Exp $ */
 
 
 /*
@@ -1451,7 +1451,7 @@
 	indp[count - 1].extraOptions = xf86addNewOption(NULL, "CorePointer", NULL);
 	indp[count].identifier = NULL;
 	servlayoutp->inputs = indp;
-    } else {
+    } else if (!havePointer) {
 	/* This should never happen. */
 	xf86Msg(X_ERROR, "Cannot locate a core pointer device.\n");
 	return FALSE;
@@ -1473,7 +1473,7 @@
 	indp[count - 1].extraOptions = xf86addNewOption(NULL, "CoreKeyboard", NULL);
 	indp[count].identifier = NULL;
 	servlayoutp->inputs = indp;
-    } else {
+    } else if (!haveKeyboard) {
 	/* This should never happen. */
 	xf86Msg(X_ERROR, "Cannot locate a core keyboard device\n");
 	return FALSE;


